### DISCUSSION TITLE
Player market value growth/decline

### DISCUSSION THREAD

**Discussion by R4nd0** [fabfischer/kickbase-plus#47] [Q&A] [answered]
I enjoy both, kickbase+ and kickbaseAutoOffer. 

I managed to customize kickbaseAutoOffer, however, I cannot see or figure out how to get the market value growth/decline trough the api. The absolute growth/decline in cash from yesterday to today would give new automated trading strategy possibilities.
fabfischer solved it and must have put the logic to PlayerMarketValueTrend.vue (Line83) but I lack the skills transfering his work to py code that works with the py api. 

Any support or solution is very much appreciated.

**Comment by fabfischer** [comment, marked as answer]
Hi @R4nd0 ,

it is really simple. You just need the statistic data of a player (`marketValues`) from the kickbase API (e.g. `https://api.kickbase.com/leagues/XXXX/players/2279/stats`).

This is an object of a player, and the property `marketValues` holds the market values for the last year. You can reverse this array and the first item in it would be the most recent market value. You can compare the current market value with the market value of the day before and you have the growth/shrinkage of the market value.

example of a player json:
```json
# https://api.kickbase.com/leagues/XXXX/players/2279/stats
{
    [...]
    "marketValues": [
        {
            "d": "2021-09-14T22:00:00Z",
            "m": 500000
        },
        [...]
        {
            "d": "2022-09-12T22:00:00Z",
            "m": 10486151
        },
        {
            "d": "2022-09-13T22:00:00Z",
            "m": 10488297
        }
    ]
    [...]
}
```

my code, changed to be more pseudo code to clarify logic:

```javascript
# src/helper/helper.js:51
const getMarketValueGrowth = (id) => {
    const allPlayers = OBJECT_WITH_ALL_PLAYERS
    const playerData = allPlayers[id]

    // we need to reverse the array, 
    // because the first element actually represents the oldest market value, not the newest
    const marketValues = allPlayers[id].marketValues.reverse()

    if (marketValues[marketValues.length - 1] && marketValues[marketValues.length - 2]) {
        return marketValues[marketValues.length - 1].m - marketValues[marketValues.length - 2].m
    }
    return null
}
```

**Comment by R4nd0** [comment]
Hi @fabfischer, thanks a lot! This was very helpful and brought me to figure out the rest on my own :) Thanks again!
